### ISSUE TITLE
Support fungible assets in `buy`

### ISSUE THREAD

**Issue by grabbou** [metaplex-foundation/js#482]
Could it be possible for `buy()` to support `tokens` and/or `price` to support purchasing fungible assets? For example, there's 5 assets listed for a total price of 5. It should be possible to buy a given number of assets (price would be automatically calculated in such case).

Right now, alternative approach is to create bid and execute sale separately, which is pretty much what `buy` does in a single transaction.

**Comment by staccDOTsol** [comment]
Yes, it is possible to support fungible assets in `buy()` by allowing the user to specify the number of tokens they want to purchase instead of a fixed price. The price can then be automatically calculated based on the current market price of the token. 

To implement this, you would need to modify the `buy()` function to accept a `tokenAmount` parameter in addition to the `price` parameter. The `tokenAmount` parameter would represent the number of tokens the user wants to purchase. 

You would then need to calculate the total price of the purchase by multiplying the `tokenAmount` by the current market price of the token. This can be done by querying the blockchain for the current market price of the token using a price oracle or other similar mechanism.

Once you have calculated the total price, you can then execute the purchase transaction as usual, using the `buy()` function to create and execute the transaction in a single step.

This approach would allow users to purchase a given number of tokens at the current market price, without having to manually calculate the price themselves or create a separate bid and execute sale transaction.
